src/coins.h

```diff
@@ -474,6 +482,14 @@ class CCoinsViewCache : public CCoinsViewBacked
     //! See: https://stackoverflow.com/questions/42114044/how-to-release-unordered-map-memory
     void ReallocateCache();
 
+    /**
+     * Reserve enough space in the cache so the underlying unordered_map will
+     * not have to rehash unless capacity is exceeded.
+     */
+    void Reserve(size_t capacity) {
```

Comment:
NIt: can you please reformat the change with latest clang-format?
nit2: this seems to belong closer to `GetCacheSize`, one reserves, the other returns the actual size